packages/core/src/cloud-wallet.ts

```diff
@@ -2,13 +2,302 @@ import {
+      { name: 'AES-GCM' },
+      false,
+      ['decrypt']
+    );
+
+    const decryptedBuffer = await crypto.subtle.decrypt(
+      { name: 'AES-GCM', iv: ivData },
+      key,
+      encryptedData
+    );
+
+    const decrypted = new TextDecoder().decode(decryptedBuffer);
+
+    return decrypted;
+  } catch (error) {
+    throw new Error('Decryption failed: Invalid key or corrupted data');
+  }
+}
+
+/**
+ * Initializes the KeyMappingVault using biometric data
+ * @param edvUrl URL for the edv
+ * @param authKey Auth key for the edv
+ * @param biometricData User's biometric data
+ * @param identifier User's identifier (email, phone number, etc.)
+ * @returns Initialized EDV service
+ */
+export async function initializeKeyMappingVault(
+  edvUrl: string,
+  authKey: string,
+  biometricData: any,
+  identifier: string
+): Promise<any> {
+  const {
+    hmacKey,
+    agreementKey,
+    verificationKey
+  } = await deriveKeyMappingVaultKeys(biometricData, identifier);
+
+  const keyMappingEdvService = new EDVService();
+  await keyMappingEdvService.initialize({
+    hmacKey,
+    agreementKey,
+    verificationKey,
+    edvUrl,
+    authKey
+  });
+
+  return keyMappingEdvService;
+}
+
+/**
+ * Enrolls a user by creating necessary vaults and keys
+ * @param edvUrl URL for the edv
+ * @param authKey Auth key for the edv
+ * @param biometricData Biometric data from provider
+ * @param identifier User's identifier (email, phone number, etc.)
+ * @returns The master key and mnemonic for backup
+ */
+export async function enrollUserWithBiometrics(
+  edvUrl: string,
+  authKey: string,
+  biometricData: any,
+  identifier: string
+): Promise<{ masterKey: string; mnemonic: string }> {
+  const keyMappingEdv = await initializeKeyMappingVault(
+    edvUrl,
+    authKey,
+    biometricData,
+    identifier
+  );
+
+  const { mnemonic, masterKey } = await generateCloudWalletMasterKey();
+  const { key: encryptionKey, iv } = await deriveBiometricEncryptionKey(biometricData, identifier);
+  const encryptedMasterKey = await encryptMasterKey(masterKey, encryptionKey, iv);
+
+  await keyMappingEdv.insert({
+    document: {
+      content: {
+        id: identifier,
+        type: KEY_MAPPING_TYPE,
+        encryptedKey: encryptedMasterKey
+      }
+    }
+  });
+
+  return { masterKey, mnemonic };
+}
+
+/**
+ * Authenticates a user with biometric data and identifier
+ * @param edvUrl URL for the edv
+ * @param authKey Auth key for the edv
+ * @param biometricData Biometric data from the provider
+ * @param identifier User's identifier (email, phone number, etc.)
+ * @returns The decrypted master key for CloudWalletVault
+ */
+export async function authenticateWithBiometrics(
+  edvUrl: string,
+  authKey: string,
+  biometricData: any,
+  identifier: string
+): Promise<string> {
+  const keyMappingEdv = await initializeKeyMappingVault(
+    edvUrl,
+    authKey,
+    biometricData,
+    identifier
+  );
+
+  const result = await keyMappingEdv.find({
```

Comment:
maybe not for this PR but we should have some local cache here, as this document wont change once created

Reply:
Will keep this in mind.